### PR TITLE
urw-base35-fonts: New port, version 2017-07-27

### PR DESCRIPTION
**Pull request by lemzwerg**
###### Type(s)

- [ ] enhancement

###### Tested on

macOS 10.7.5 11G63
Xcode 4.6.3 4H1503 

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?

A follow-up to the withdrawn pull request https://github.com/macports/macports-ports/pull/2658

### CODE REVIEW

File: x11/urw-base35-fonts/Portfile [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+description         URW base35 fonts in various formats
+long_description    "Recent Ghostscript versions come with a new set\
+                    of URW fonts that have additional support for\
+                    the Greek and Cyrillic scripts (thus superseding\
+                    the fonts contained in the 'urw-fonts' port).\
+                    Ghostscript distributes the URW fonts in Type1\
+                    format; in addition to these files, this port\
+                    also contains the corresponding Type1 metrics\
+                    files (AFM), together with TrueType (TTF) and\
+                    OpenType (OTF) versions of the fonts."
```
Comment (mf2k) [inline x11/urw-base35-fonts/Portfile]: We don't normally quote the long_description. Please remove them.

File: x11/urw-base35-fonts/Portfile [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+homepage            https://www.ghostscript.com
+master_sites        "https://git.ghostscript.com/?p=urw-core35-fonts.git;a=snapshot;h=${git_version};sf=tgz;dummy="
```
Comment (mf2k) [inline x11/urw-base35-fonts/Portfile]: I have never seen `master_sites` quoted before. Please remove the quotes here too.

File: x11/urw-base35-fonts/Portfile (hunk 1 above)
Comment (lemzwerg) [reply x11/urw-base35-fonts/Portfile]: It's not possible to remove the quotes.  The reason is the `; in`, which makes Tcl interpret the `in` as a command.  Rewording the paragraph just for the sake of avoiding quotes looks silly to me (it might be bad English since it isn't my mother tongue, but that would be another issue :-)

File: x11/urw-base35-fonts/Portfile (hunk 2 above)
Comment (lemzwerg) [reply x11/urw-base35-fonts/Portfile]: Similar issue.  If you do `git grep 'master_sites.*;'` you can see that it is necessary to either use `\;` instead of `;` or quote the whole string.  I like the latter more.

Comment (pmetzger) [reply x11/urw-base35-fonts/Portfile]: @lemzwerg Please don't resolve conversations on behalf of project members asking for changes. If @mf2k feels comfortable with your responses, he'll decide to accept.

Comment (lemzwerg) [reply x11/urw-base35-fonts/Portfile]: Ah, ok.  A misunderstanding.  I thought it is my job to press the 'resolve' button to indicate that I've sufficiently answered a question.

File: x11/urw-base35-fonts/Portfile (hunk 1 above)
Comment (mf2k) [reply x11/urw-base35-fonts/Portfile]: Got it. I didn't know that.

Comment (pmetzger) [reply x11/urw-base35-fonts/Portfile]: Does this then end up embedding long chunks of spaces from the start of the line into the description?

**Comment by mojca** [comment]
If upstream calls it `urw-core35-fonts`, I believe we should call the package the same. I did not notice the difference of `base` vs. `core` earlier. We should also make the `urw-base` name more clear in some way (I did not notice it was some random Russian fork) and eventually get rid of it, but that's an independent thing. I would check in what way precisely Lilypond depends on those fonts. Again, unrelated to this particular PR.

Regarding the livecheck: I believe (but please don't hold my word for it) that github PortGroup does something in the sense of what you want to do. I would need some time to decipher what the livecheck says, but I noticed that it uses
```
 livecheck.type regexm
```
so basically a multiline regexp. I (blindly) assume that it takes the first or the last match on the page when fetching from a particular commit rather than a particular tag.

**Comment by lemzwerg** [comment]
OK, will rename to `urw-core35-fonts`, see [new pull request]().

Regarding the difference between `urw-fonts` and `urw-core35-fonts`: I already mention in the port description that the latter supersedes the former – isn't that sufficient?

The old `lilypond` distribution depends on exact filenames from `urw-fonts`, while `lilypond-devel` only relies on TeX Gyre fonts currently; however, it has an additional option that I want to add to the port so that it also relies on `urw-core35-fonts` (it needs exact `*.otf` filenames) to support Greek and Cyrillic out of the box.  With 'out of the box' I mean that you can simply type Greek or Cyrillic in a lilypond input file without the need to specially select a Greek or Cyrillic font.  In particular, to build the complete documentation you need support for those two scripts since some examples use it.

And thanks for the livecheck explanations.  Right now, I won't add them :-)

**Comment by mojca** [comment]
Why a new pull request instead of changing the old one? If you end up with 10 pull requests for the same port, it's nearly impossible for someone to track the old discussion.

Btw: TeX Gyre authors would gladly include the Cyrillic part. The problem is that they realised the font quality was low and they did not find an expert that could help them improve the quality of the glyphs, so they removed the Cyrillic glyphs at some point. I didn't check these fonts at all, but if they contain Cyrillic glyphs of better (sufficient) quality, maybe they could be added to TeX Gyre as well? It might be worth asking the Polish font developers in any case.

If `lilypond-devel` no longer needs those older fonts, maybe we don't need to worry – once 2.20 is out, we would be able to drop those fonts anyway. (Except ... they didn't make a stable version since almost five years, which doesn't fill me with so much hope.)

**Comment by lemzwerg** [comment]
I submitted a new pull request because the name of the Portfile changed.  Hopefully, this won't happen again :-/

Yes, I know that the TeX Gyre fonts lack Cyrillic and Greek for the reasons you mention.  If you have contact to the Polish font team, please drop them a mail!

`lilypond-devel` won't need the `urw-fonts` port, correct.  And you are right that a new stable lilypond version is still missing.  David Kastrup is already cherry-picking commits into the 2.20 branch; maybe we get something within a few months.  BTW, the last development version (2.19.82) is cut from the 2.20 branch – given the maturity of this version it maybe makes sense to rename `lilypond-devel` to `lilypond-2.20` so that more people use it who would be otherwise afraid of the 'devel' part in the port name.  However, it will take some time until I'm there.

**Comment by mojca** [comment]
There is absolutely no need to open a new pull request just because the port name has changed. It is easy to force push commits that just renames the port (and potentially edit the title of the pull request if needed).
